app/streams/publishing_api/consumer.rb

```diff
@@ -1,7 +1,7 @@
 module PublishingAPI
   class Consumer
     def process(message)
-      MessageProcessorJob.perform_later(message.payload) if valid_routing_key?(message)
+      MessageProcessorJob.perform_later(message.payload, message.delivery_info.routing_key) if valid_routing_key?(message)
```

Comment:
Not sure about passing the routing key as a second parameter.
It would be enough with inserting this code in the Consumer?

```ruby
def process(message)
  Monitor.Messages.process(message)
  MessageProcessorJob.perform_later(message.payload) if valid_routing_key?(message)
end
```